docs/asch_sdk_api_en.md

```diff
@@ -0,0 +1,622 @@
+/* output:
+[
+  {
+    tid: "50e062f25946d220b924cb5ec6e52e260e44c9417d9f3c8ea041b704e06895f7",
+    senderId: "AFnwUuET2XddPtqpFb2ns78CQEqc7KZ6vD",
+    recipientId: "asdasdasd",
+    currency: "CCTime.XCT",
+    amount: "100000000",
+    t_timestamp: 38660145,
+    t_type: 3,
+    t_height: 93953
+  },
+  {
+    tid: "f15ce92add809b4a132936d514dce7fa7bdc15e850e7c026a001625b48595af3",
+    senderId: "AFnwUuET2XddPtqpFb2ns78CQEqc7KZ6vD",
+    recipientId: "asdasd",
+    currency: "CCTime.XCT",
+    amount: "100000000",
+    t_timestamp: 38660096,
+    t_type: 3,
+    t_height: 93948
+  }
+]
+*/
+```
+
+## 4. Routing
+
+- `path` Path
+- `handler` Http Request Handler (async)
+
+### 4.1 app.route.get(path, handler)
+
+> Register a `get` type `http` request handler
+
+### 4.2 app.route.post(path, handler)
+
+> Register a `post` type  `http` request handler
+
+### 4.3 app.route.put(path, handler)
+
+> Register a `put` type `http` request handler
+
+## 5. Cost pool
+
+### 5.1 app.feePool.add(currency, amount)
+
+- `currency` Currency
+- `amount` Amount
+
+> No return value
+> The specified asset (currency) is added to the fee pool (the amount is evenly distributed to all accountants at the end of each round).
+
+Example:
+
+```
+app.feelPool.add('XAS', '10000000')
+```
+
+## 6. Icrement ID
+
+### 6.1 app.autoID.get(name)
+
+- `name` Id Type
+
+> Returns a `String`
+> Get the highest Id of a type
+
+### 6.2 app.autoID.increment(name)
+
+- `name` Id Type
+
+> Returns a `String`
+> Increment the Id of the specified type by 1 and return the updated value as a string. This is equivalent to the `++i` operation in javascript.
+
+Example:
+
+```
+const AID = 'article_id'
+app.autoID.get(AID) === '0'
+app.autoID.increment(AID) === '1'
+app.autoID.get(AID) === '1'
+```
+
+## 7. 日志
+
+### 7.1 app.logger.setLevel(level)
+### 7.2 app.logger.log()
+### 7.3 app.logger.trace()
+### 7.4 app.logger.debug()
+### 7.5 app.logger.info()
+### 7.6 app.logger.warn()
+### 7.7 app.logger.error()
+
+Example
+
+```
+app.logger.setLevel('debug')
+app.logger.setLevel('info')
+
+logger.log('hello');
+logger.trace('hello', 'world');
+logger.debug('hello %s',  'world', 123);
+logger.info('hello %s %d',  'world', 123, {foo:'bar'});
+logger.warn('hello %s %d %j', 'world', 123, {foo:'bar'});
+logger.error('hello %s %d %j', 'world', 123, {foo:'bar'}, [1, 2, 3, 4], Object);
+```
+
+## 8. Tools
+
+### 8.1 app.validate(type, value)
+
+- `type` The datatype that should be verified
+- `value` Data value which will be verified
+
+> Verify that a data is matches the specification. If not, throw an exception.
+
+Example:
+
+```
+app.validate('amount', '10000') // pase
+app.validate('amount', 10000) // throws
+app.validate('amount', 'abc') // throws
+app.validate('amount', '1e10') // throws
+```
+
+### 8.2 app.registerContract(type, name)
+
+- `type` Contract Type
+- `name` The name of contract
+
+> No return value
+> 为合约注册一个数字类型, 未注册的合约无法被外部调用
```

Comment:
Why Chinese here?

Reply:
I couldn't translate it in a meaningful way.